services/src/datasets/external/nfdi/mod.rs

```diff
@@ -458,25 +523,35 @@ impl MetaDataProvider<OgrSourceDataset, VectorResultDescriptor, VectorQueryRecta
         Box<dyn MetaData<OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>>,
     > {
         let id: DataId = id.clone().into();
-        let (_, md) = self.dataset_info(&id).await.map_err(|e| {
+
+        let aruna_dataset_ids = self.get_dataset_info(&id).await.map_err(|e| {
             geoengine_operators::error::Error::DatasetMetaData {
                 source: Box::new(e),
             }
         })?;
 
-        let object = self.get_single_file_object(&id).await.map_err(|e| {
+        let meta_data = self.get_metadata(&aruna_dataset_ids).await.map_err(|e| {
             geoengine_operators::error::Error::DatasetMetaData {
                 source: Box::new(e),
             }
         })?;
 
-        match md.data_type {
+        let aruna_data_object = self
+            .get_single_file_object(&aruna_dataset_ids)
+            .await
+            .map_err(|e| geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(e),
+            })?;
+
+        match meta_data.data_type {
             DataType::SingleVectorFile(info) => {
-                let result_descriptor = Self::create_vector_result_descriptor(md.crs.into(), &info);
+                let result_descriptor =
```

Comment:
please set the temporal bound of the ResultDescriptor if i is set in the VectorInfo. Can we add (optional) spatial bounds to VectorInfo to set the spatial bounds of the ResultDescriptor?

Reply:
92da6f78a46dc3666f36e9a15225addf43a6c333

I added both time and spatial bounds to VectorInfo. Time was also necessary, because temporalExtend are the fields used for time, not the actual time itself. Similar to RasterInfo, VectorInfo is almost the same as a VectorResultDescriptor now, but I won't do that now.